models/scheduled_pull_request_merge.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2021 Gitea. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"code.gitea.io/gitea/modules/timeutil"
+)
+
+// ScheduledPullRequestMerge represents a pull request scheduled for merging when checks succeed
+type ScheduledPullRequestMerge struct {
+	ID          int64              `xorm:"pk autoincr"`
+	PullID      int64              `xorm:"BIGINT"`
+	DoerID      int64              `xorm:"BIGINT"`
+	Doer        *User              `xorm:"-"`
+	MergeStyle  MergeStyle         `xorm:"varchar(50)"`
+	Message     string             `xorm:"TEXT"`
+	CreatedUnix timeutil.TimeStamp `xorm:"created"`
+}
+
+// ScheduleAutoMerge schedules a pull request to be merged when all checks succeed
+func ScheduleAutoMerge(doer *User, pullID int64, style MergeStyle, message string) (err error) {
+	sess := x.NewSession()
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+	defer sess.Close()
+
+	// Check if we already have a merge scheduled for that pull request
+	exists, _, err := getScheduledPullRequestMergeByPullID(sess, pullID)
+	if err != nil {
+		return
+	} else if exists {
```

Comment:
```suggestion
	}
	if exists {
```

Reply:
It's about limit variable scope ... refactored more ...